src/zjs_console.c

```diff
@@ -14,50 +14,78 @@
 #define STDOUT_PRINT(s, ...) PRINT(s, __VA_ARGS__)
 #endif
 
+#define IS_NUMBER 0
+#define IS_INT    1
+#define IS_UINT   2
+
+int is_int(jerry_value_t val) {
+    int ret = 0;
+    double n = jerry_get_number_value(val);
+    ret = (n - (int) n == 0);
+    if (ret) {
+        // Integer type
+        if (n < 0) {
+            return IS_INT;
+        } else {
+            return IS_UINT;
+        }
+    } else {
+        return IS_NUMBER;
+    }
+}
+
+static jerry_value_t do_print(const jerry_value_t function_obj,
+                              const jerry_value_t this,
+                              const jerry_value_t argv[],
+                              const jerry_length_t argc,
+                              FILE* out)
+{
+    int i;
+    for (i = 0; i < argc; ++i) {
+        if (jerry_value_is_string(argv[i])) {
+            jerry_size_t jlen = jerry_get_string_size(argv[0]);
+            if (jlen > MAX_STR_LENGTH) {
+                ERR_PRINT("console string is too long, max length=%u\n", MAX_STR_LENGTH);
+                return ZJS_UNDEFINED;
+            }
+            char buffer[jlen + 1];
+            int wlen = jerry_string_to_char_buffer(argv[0], (jerry_char_t *)buffer, jlen);
+            buffer[wlen] = '\0';
+
+            fprintf(out, "%s\n", buffer);
```

Comment:
I suggest you print the arguments separated by spaces instead of newlines; this is what I expect at least and what it seems to do in e.g. Chrome Inspector. (Then add a newline at the end.)